### DISCUSSION TITLE
Cannot override default admin credentials

### DISCUSSION THREAD

**Discussion by mindrunner** [emqx/emqx#7573] [Q&A] [answered]
For my deployment, I need to get rid of the unsecure default credentials. I followed the guides and read quite a bit of the bugtracker, seeing people have similar problems. What is the best practice here?

Goal: 
- no `admin`:`public` user login possible.
- no default `admin`:`public` application credentials

Steps to reproduce:
Start container with following env-variabls:
```
      EMQX_AUTH__USER__1__USERNAME: iamsuperuser
      EMQX_AUTH__USER__1__PASSWORD: mysupersecurepassword
      EMQX_MANAGEMENT__DEFAULT_APPLICATION__ID: iamsuperuser
      EMQX_MANAGEMENT__DEFAULT_APPLICATION__SECRET: mysupersecurepassword
      EMQX_MANAGEMENT__APPLICATION__DEFAULT_SECRET: mysupersecurepassword
```

After that, login is still possible with default credentials. Furthermore, the credentials set in environment are not used by emqx.

What am I missing here?

**Comment by zmstone** [comment]
Config and env var are only used to bootstrap the record in database. If the database is already initiated, you’ll need to reset from `emqx_ctl admins` command

**Comment by mindrunner** [comment]
I am not sure if I do understand correctly. I am doing this with the vanilla docker image with no volumes mounted (e.g. clean, no persistence, no already initialized databases)

**Reply by techmantel** [reply]
@mindrunner Did you get the mqtt client authorization working using those environmentals `EMQX_AUTH__USER__1__XXX`?
Tried it but my client only gets a not authorized response 🤔

**Reply by techmantel** [reply]
Ok, figured that it works when I enable the `emqx_auth_mnesia` plugin, but how can I enable it from environment vars?

**Reply by mindrunner** [reply]
I am using the helm chart and enable pugins via `values.yml`

```
emqxLoadedModules: |
  {emqx_mod_topic_metrics, true}. {emqx_mod_trace, true}. {emqx_mod_slow_subs, true}.
emqxLoadedPlugins: |
  {emqx_management, true}. {emqx_recon, false}. {emqx_retainer, false}. {emqx_dashboard, true}. {emqx_telemetry, true}. {emqx_rule_engine, true}. {emqx_bridge_mqtt, false}. {emqx_auth_mnesia, true}. {emqx_prometheus, true}.
```

**Reply by techmantel** [reply]
Sorry, I'm not familiar with helm. How will that end up in during the actual deployment? I assume it alters the config files directly, or?

**Reply by mindrunner** [reply]
If you want to load per environment variable (e.g. in compose setups), you can use:

```
EMQX_LOADED_PLUGINS: "emqx_recon,emqx_retainer,emqx_management,emqx_dashboard,emqx_auth_mnesia"
EMQX_LOADED_MODULES: "emqx_mod_topic_metrics"
```

**Comment by zmstone** [comment, marked as answer]
Hi @mindrunner 
Sorry that I misunderstood the issue.

EMQX default user is configurable as:

```
EMQX_dashboard__default_user__login=iamsuperuser
EMQX_dashboard__default_user__password=mysupersecurepassword
```

**Reply by mindrunner** [reply]
Cool, that works like a charm! :) Thank you so much.
